ENIGMAsystem/SHELL/Widget_Systems/Win32/gtklike.h

```diff
@@ -113,14 +117,18 @@ struct gtkl_table: gtkl_container
   {
     x=xnew, y=ynew, w=dwid, h=dhgt;
     int cw = dwid / gsx, ch = dhgt / gsy;
-    if (ch<gsy) ch=gsy;
-    if (cw<gsx) cw=gsx;
 
-    int givex[gsx][gsy], givey[gsx][gsy];
+    if (ch<gsy) ch=gsy; 
+    if (cw<gsx) cw=gsx;
+    
+    std::vector<std::vector<int>> givex(gsx), givey(gsx);
     // Make all occupied cells yield max give
-    for (int i = 0; i < gsx; i++)
+    for (int i = 0; i < gsx; i++) {
+      givex[i].resize(gsy);
```

Comment:
This is sort of playing with fire. But since we only access the current array in this loop body, this is fine. (I'd  feel safer if you initialized the matrices this way up-front, but fine.)

Reply:
Yeah I was trying to tell him about contiguity and row/column major iteration to make it cache friendly, but I didn't even know how to read it myself, so I told him to wait for you on this one.